File: nexustrace-backend/app/schemas/case.py
from pydantic import BaseModel
from typing import Optional, Literal, List
from datetime import datetime

class CaseCreate(BaseModel):
    name: str
    description: Optional[str] = None
    priority: Optional[Literal["low", "medium", "high", "critical"]] = "medium"
    tags: Optional[List[str]] = []

class CaseResponse(BaseModel):
    case_id: str
    name: str
    description: Optional[str] = None
    created_at: int # Neo4j timestamp to int
    status: Optional[str] = "open"
    priority: Optional[str] = "medium"
    tags: Optional[List[str]] = []
    evidence_count: Optional[int] = 0

class CaseUpdate(BaseModel):
    status: Optional[Literal["open", "in_progress", "closed"]] = None
    name: Optional[str] = None
    description: Optional[str] = None
    priority: Optional[Literal["low", "medium", "high", "critical"]] = None
    tags: Optional[List[str]] = None

File: nexustrace-frontend/types/case.ts
export interface Case {
  id?: string;
  case_id?: string;
  title?: string;
  name?: string;
  description: string;
  status?: "open" | "closed" | "in_progress";
  priority?: "low" | "medium" | "high" | "critical";
  tags?: string[];
  created_at: string | number;
  updated_at?: string;
  owner_id?: string;
  evidence_count?: number;
}

export interface CreateCaseRequest {
  name: string;
  description: string;
  priority?: "low" | "medium" | "high" | "critical";
  tags?: string[];
}

export interface Evidence {
  evidence_id: string;
  filename: string;
  file_type: string;
  created_at: string | null;
}

export interface TimelineEvent {
  id: string;
  timestamp: string;
  event_type: string;
  description: string;
  source: string;
  entities: string[];
  risk_score?: number;
}

export interface Entity {
  id: string;
  name: string;
  type: "person" | "organization" | "location" | "device" | "email" | "ip" | "phone" | "other";
  mentions: number;
  risk_score?: number;
  properties?: Record<string, string>;
}

export interface PrioritizedLead {
  id: string;
  entity: string;
  entity_type: string;
  risk_score: number;
  reason: string;
  connections: number;
  last_seen: string;
  risk_breakdown?: Record<string, number>;
  top_risk_drivers?: string[];
  counterfactual_explanations?: string[];
}
